src/AzureDevOps.DataIngestor.Sdk/Ingestors/CommitIngestor.cs

```diff
@@ -0,0 +1,130 @@
+﻿using AzureDevOps.DataIngestor.Sdk.Clients;
+using AzureDevOps.DataIngestor.Sdk.Entities;
+using AzureDevOps.DataIngestor.Sdk.Util;
+using EntityFrameworkCore.BulkOperations;
+using Microsoft.Extensions.Logging;
+using Microsoft.TeamFoundation.Core.WebApi;
+using Microsoft.TeamFoundation.SourceControl.WebApi;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace AzureDevOps.DataIngestor.Sdk.Ingestors
+{
+    public class CommitIngestor : BaseIngestor
+    {
+        private readonly ILogger logger;
+        private readonly VssClient vssClient;
+        private readonly string sqlConnectionString;
+        private readonly IEnumerable<string> projectNames;
+
+        public CommitIngestor(VssClient vssClient, string sqlConnectionString, IEnumerable<string> projectNames, ILogger logger)
+        {
+            this.logger = logger;
+            this.vssClient = vssClient;
+            this.sqlConnectionString = sqlConnectionString;
+            this.projectNames = projectNames;
+        }
+
+        public override async Task RunAsync()
+        {
+            // Get projects from Azure DevOps
+            List<TeamProjectReference> projects = await this.vssClient.ProjectClient.GetProjectsAsync(this.projectNames);
+
+            // For each project, retrieve and ingest commit data
+            foreach (TeamProjectReference project in projects)
+            {
+                List<GitRepository> reposFromProject = await this.vssClient.GitClient.GetRepositoriesWithRetryAsync(project.Name);
+
+                foreach (GitRepository repo in reposFromProject)
+                {
+                    this.logger.LogInformation($"Retrieve and ingest commits");

```

Comment:
>Retrieve [](http://example.com/codeflow?start=49&length=8)

Maybe log the project name if it's not logged somewhere else already.

Reply:
It's logged in calls before but does not hurt